extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -257,6 +273,45 @@ export class SubscriptionService {
     return this;
   }
 
+  /**
+   * Delegates authentication to viewer
+   */
+  delegateAuthToViewer_() {
+    const serviceIds = ['local'];
+    const currentProductId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getProductId(),
+        'Product id is null'
+    ));
+    const publicationId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getPublicationId(),
+        'Publication id is null'
+    ));
+    const origin = getWinOrigin(this.ampdoc_.win);
+    this.platformStore_ = new PlatformStore(serviceIds);
+    this.viewer_.sendMessageAwaitResponse('auth', dict({
+      'publicationId': publicationId,
+      'productId': currentProductId,
+      'origin': origin,
+    })).then(entitlementData => {
+      const authData = (entitlementData || {})['authorization'];
+      dev().assert(authData, 'authorization is not defined');
+      const decodedData = this.jwtHelper_.decode(authData);
+      if (decodedData['aud'] != origin) {
+        user().error(TAG, 'Audience does not match');
+      }
+      if (Date.now() - decodedData['exp'] < 1000 * 60) {
+        user().error(TAG, 'Payload about to expire');
+      }
+      const entitlements = decodedData['entitlements'];
+      user().assert(entitlements[0], 'No entitlements found');
+      const entitlement = Entitlement.parseFromJson(entitlements[0]);
+      // Viewer authorization is redirected to use local platform instead.
+      this.platformStore_.resolveEntitlement('local', entitlement);
+    }, reason => {
+      throw user().createError('Viewer authorization failed', reason);
```

Comment:
So, in a non-viewer scenario, you'd still resolve entitlements with `empty` entitlement, even when platform fails. Right?